django/contrib/postgres/apps.py

```diff
@@ -15,6 +15,10 @@ class PostgresConfig(AppConfig):
     def ready(self):
         # Connections may already exist before we are called.
         for conn in connections.all():
+            if conn.vendor == 'postgresql':
+                conn.introspection.data_types_reverse.update({
+                    3802: 'django.contrib.postgresql.fields.JSONField',
+                })
             if conn.connection is not None:
```

Comment:
This could be indented inside the `if conn.vendor` to avoid a redundant check.

Reply:
```
if conn.vendor == 'postgresql':
    conn.introspection.data_types_reverse.update({
    ...
    if conn.connection is not None:
```
(register_type_handlers() also checks `conn.vendor == 'postgresql'` so that avoids that same check for the non-postgres case)